test/integration/things-media.js

```diff
@@ -0,0 +1,201 @@
+const CONFIG = require('config')
+const __ = CONFIG.universalPath
+const _ = __.require('lib', 'utils')
+
+require('should')
+
+const breq = require('bluereq')
+const get = (url) => breq.get(url).then(_.property('body'))
+const post = (url, body) => breq.post(url, body).then(_.property('body'))
+const put = (url, body) => breq.put(url, body).then(_.property('body'))
+const delete_ = (url) => breq.delete(url).then(_.property('body'))
+
+const endpoint = CONFIG.server.url() + '/media'
+const mediaNewURL = require('../fixtures/media-new-url.json')
+const mediaNewIPFS = require('../fixtures/media-new-ipfs.json')
+const mediaNewBLOB = '../fixtures/SDvision_MN00097-intermediate-DMs1p36833449a6size0_0005.jpg'
+
+describe('/media', function () {
+  describe('POST url', function () {
+    it('should return metadata with journal id, assetUrl', function (done) {
+      post(endpoint, mediaNewURL)
+      .then(function (body) {
+        body.ok.should.equal(true)
+        _.isUuid(body.id).should.equal(true)
+        _.isNonEmptyString(body.assetUrl)
+        done()
+      })
+    })
+    it('should return a 400 on invalid doc', function (done) {
+      const invalidNewDoc = {
+        so: 'much',
+        invalid: true
+      }
+      breq.post(endpoint, invalidNewDoc)
+      .then(function (res) {
+        res.statusCode.should.equal(400)
+        done()
+      })
+    })
+  })
+  describe('POST ipfs metadata', function () {
+    it('should return metadata with journal id, assetUrl, ipfs hash and mimetype', function (done) {
+      post(endpoint, mediaNewIPFS)
+      .then(function (body) {
+        body.ok.should.equal(true)
+        _.isUuid(body.id).should.equal(true)
+        body.mimetype.should.equal('image/jpeg')
+        body.ipfs.should.equal('QmVXmy59f5QqKDGyhpHbRiGnba5SfHaooDPCWhd8Xtgwz2')
+        _.isNonEmptyString(body.assetUrl).should.equal(true)
+        done()
+      })
+    })
+    it('should return a 400 on invalid doc', function (done) {
+      const invalidNewDoc = {
+        so: 'much',
+        invalid: true
+      }
+      breq.post(endpoint, invalidNewDoc)
+      .then(function (res) {
+        res.statusCode.should.equal(400)
+        done()
+      })
+    })
+  })
+  describe('POST image', function () {
```

Comment:
The [content disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) header of an HTTP POST upload contains a value `filename`, which we could sanitizingly infer a `name` from. Then metadata is detected to a broad extent.

Asking how to prepare this for compatibility with IIIF from here is left to succeeding iterations on the codebase. I may safely conclude KISS applies for now.

Still, this involves correct handling of `multipart`.